src/main/java/com/synopsys/integration/detect/workflow/blackduck/font/DetectFontInstaller.java

```diff
@@ -35,21 +42,33 @@ public void installFonts(File targetDirectory) {
             String location = artifactResolver.resolveArtifactLocation(ArtifactoryConstants.ARTIFACTORY_URL, ArtifactoryConstants.FONTS_REPO, ArtifactoryConstants.FONTS_PROPERTY, "", "");
             File fontZipFile = downloadZipFile(location, targetDirectory);
             expandZipFile(fontZipFile, targetDirectory);
+            installedToolManager.saveInstalledToolLocation(FONTS_ZIP_KEY, targetDirectory.getAbsolutePath());
+
             String zipFilePath = fontZipFile.getAbsolutePath();
             logger.debug("Deleting zip file {}", zipFilePath);
             boolean zipDeleted = FileUtils.deleteQuietly(fontZipFile);
             if (zipDeleted) {
                 logger.debug("Successfully deleted zip file {}", zipFilePath);
             }
-        } catch (IOException | IntegrationException | ArchiveException ex) {
-            logger.error(String.format("Failed to load font files into %s", targetDirectory.getAbsolutePath()));
-            logger.error("Cause: ", ex);
+        } catch (IOException | IntegrationException | ArchiveException e) {
+            logger.debug("Attempting to download fonts from previous install.");
+            installedToolLocator.locateTool(FONTS_ZIP_KEY).ifPresent(fonts ->
+            {
+                try {
+                    FileUtils.copyDirectory(fonts, targetDirectory);
+                } catch (IOException ex) {
+                    logger.error(String.format("Failed to load font files into %s", targetDirectory.getAbsolutePath()));
+                    logger.error("Online install exception cause: ", e);
+                    logger.error("Local install exception: ", ex); //TODO- should we log exceptions for both online and local installs?
+                }
+            });
```

Comment:
I like the idea of logging both errors if both attempts failed.
A total nit, but seems like kind of a big catch clause.

Reply:
agreed, the catch clause is gross I'm just not sure how to improve it...I could extract the block within the ifPresent to a private method?

Reply:
ah. Definitely not worth holding up the merge/release for that in my opinion.